parser-by-hand/Main.java

```diff
@@ -0,0 +1,160 @@
+            System.out.print("Successfully matched " + token + "\n");
+            ts.advance();
+        }
+        else {
+            String errorMessage = String.format("Expected token %s, but got %s.", token, ts.peek());
+            throw new UnexpectedTokenException(errorMessage);
+        }
+    }
+
+    private static void Stmts() {
+        switch(ts.peek()) {
+            case "textval": case "id": case "lparen": case "numberval":
+                Stmt();
+                match(ts, "nl");
+                Stmts();
+                break;
+            default: break;
+        }
+	} 
+
+	private static void Dcls() {
+        if(ts.peek() == "numberdcl" || ts.peek() == "textdcl") {
+            Dcl();
+            match(ts, "nl");
+            Dcls();
+        }
+	} 
+
+	private static void Dcl() {
+        Type();
+        match(ts, "id");
+        match(ts, "(");
+        Dclparams();
+        match(ts, ")");
+        match(ts, "=");
+        Stmt();
+	} 
+
+	private static void Stmt() {
+        if(ts.peek() == "textval") {
+            match(ts, "textval");
+        }
+        else {
+            Math();
+        }
+	} 
+
+	private static void Math() {
+        Term();
+        Followterm();
+	} 
+
+	private static void Followterm() {
+        if(ts.peek() == "+" || ts.peek() == "-") {
+            Simpleops();
+            Math();
+        }
+        
+	} 
+
+	private static void Simpleops() {
+        if(ts.peek() == "+") {
+            match(ts, "+");
+        }
+        else {
+            match(ts, "-");
+        }
+    }
+
+    private static void Term() {
+        if(ts.peek() == "id") {
+            Funccall();
+        }
+        else {
+            Val();
+        }
+	} 
+
+	private static void Val() {
+        if(ts.peek() == "(") {
+            match(ts, "(");
+            Math();
+            match(ts, ")");
+        }
+        else {
+            match(ts, "numberval");
+        }
+	} 
+
+	private static void Funccall() {
+        match(ts, "id");
+        match(ts, "(");
+        Stmtparams();
+        match(ts, ")");
+	} 
+
+	private static void Stmtparams() {
+        switch(ts.peek()) {
+            case "textval": case "id": case "lparen": case "numberval":
+                Stmt();
+                Stmtmoreparams();
+                break;
+            default: break;
+        }
+	} 
+
+	private static void Stmtmoreparams() {
+        if(ts.peek() == ",") {
+            match(ts, ",");
+            Stmt();
+            Stmtmoreparams();
+        }
+	} 
+
+	private static void Dclparams() {
+        if(ts.peek() == "numberdcl" || ts.peek() == "textdcl") {
+            Dclparam();
+            Dclmoreparams();
+        }
+	} 
+
+	private static void Dclmoreparams() {
+        if(ts.peek() == ",") {
+            match(ts, ",");
+            Dclparam();
+            Dclmoreparams();
+        }
+	}
+
+	private static void Dclparam() {
+        Type();
+        match(ts, "id");
+	}
+
+	private static void Type() {
```

Comment:
Generel kommentar/spørgsmål. Der er mange steder hvor du siger if(ts.peek == something){gør dette} else{gør dette}, men ser ikke noget sted at du smider en error hvis du ikke matcher. Altså hvis hverken den ene, anden, tredje eller nogen som helst if statement evaluerer til true, så går vi bare videre og får kun error fordi vi prøver at matche end of file ($). Altså får vi kun én fejl. Måske jeg tager fejl, men så skal jeg have det forklaret